### ISSUE TITLE
TestFramework: "Audio track position with loop points" in ResourceAudioLoopPointsTestSuite failed, Track position is not in the loop section

### ISSUE THREAD

**Issue by sihammill** [YoYoGames/GameMaker-Bugs#8245]
### Workflow Artefact URL

https://github.com/YoYoGames/GM-TestFramework/actions/runs/11610157668#artifacts

### Test Code

```
ev_step: function() {
			
	var timeSpent = (get_timer() / 1000000) - startTime; // in seconds
	
	// After 1 second, the audio should be somewhere in the loop section
	if (timeSpent >= 1) {
		
		var trackPos = audio_sound_get_track_position(sound);
		
		assert_less_or_equal(trackPos, loop_end, "Track position is not in the loop section");
		assert_greater_or_equal(trackPos, loop_start, "Track position is not in the loop section");
		
		test_end();
	}
	
},
```

### Output From The Test

```
Testsuite Name: ResourceAudioLoopPointsTestSuite
Test Name: Audio track position with loop points
description: Track position is not in the loop section
title: Assert value to be less than or equal to
stack: gml_Script_anon@1887@___struct___335@ResourceAudioLoopPointsTestSuite@ResourceAudioLoopPointsTestSuite:72
actual: 0.21342404186725616
expected: 0.20000000000000001
```

### Runtime Version

2024.1100.0.689

### Location Of The Test

https://github.com/YoYoGames/GM-TestFramework/blob/85e03a71e87b23cffb3b93b42b18081307cf2f09/projects/xUnit/scripts/ResourceAudioLoopPointsTestSuite/ResourceAudioLoopPointsTestSuite.gml#L47

### Which platform(s) are you seeing the problem on?

Windows

**Comment by toby-yoyo** [comment]
Fixed an issue in yyal where relative offsets in decoding buffers were written as byte offsets rather than frame offsets. This led to calculated track position values for compressed sounds being larger than expected in most cases.

**Comment by sihammill** [comment]
Fix verified in IDE v2024.1300.0.714  Runtime v2024.1300.0.727
Testframework test now passing as expected
